backend/src/tests.rs

```diff
@@ -217,81 +256,120 @@ mod test {
+            message:  "Summa proof of solvency for CryptoExchange".encode().into(),
+        },AddressOwnershipProof {
+        chain: "ETH".to_string(),
+        cex_address:to_checksum(&cex_addr_2, None),
+        signature:
+            ("0xb17a9e25265d3b88de7bfad81e7accad6e3d5612308ff83cc0fef76a34152b0444309e8fc3dea5139e49b6fc83a8553071a7af3d0cfd3fb8c1aea2a4c171729c1c").parse().unwrap(),
+            message:  "Summa proof of solvency for CryptoExchange".encode().into(),
+        },
+                ],
+            }
+        );
 
         // Initialize round
-        let params_path = "ptau/hermez-raw-11";
+        let params_path = "ptau/hermez-raw-17";
         let entry_csv = "../csv/entry_16.csv";
+        let mut entries: Vec<Entry<N_CURRENCIES>> = vec![Entry::init_empty(); N_USERS];
+        let mut cryptos = vec![Cryptocurrency::init_empty(); N_CURRENCIES];
+        parse_csv_to_entries::<&str, N_CURRENCIES>(entry_csv, &mut entries, &mut cryptos).unwrap();
+
+        let univariate_grand_sum_circuit =
+            UnivariateGrandSum::<N_USERS, N_CURRENCIES>::init(entries.to_vec());
+
+        let (params, pk, _) =
+            generate_setup_artifacts(K, None, &univariate_grand_sum_circuit).unwrap();
+
+        // Create a proof
+        let instances = vec![Fp::one(); 1]; // This instance is necessary to verify proof on solidity verifier.
+        let (zk_snark_proof, advice_polys, _omega) = full_prover(
+            &params,
+            &pk,
+            univariate_grand_sum_circuit.clone(),
+            &[instances.clone()],
+        );
 
-        let mst = MerkleSumTree::new(entry_csv).unwrap();
-        let mut round = Round::<4, 2, 14>::new(&signer, Box::new(mst), params_path, 1).unwrap();
-
-        let mut liability_commitment_logs = summa_contract
-            .liabilities_commitment_submitted_filter()
-            .query()
-            .await?;
-
-        assert_eq!(liability_commitment_logs.len(), 0);
-
-        // Send liability commitment transaction
-        round.dispatch_commitment().await?;
+        // Iterate entries and collect values to `[Vec[Fp]; N_POINTS]`
+        const FP_ARRAY: Vec<Fp> = Vec::new();
+        let mut user_data = [FP_ARRAY; N_POINTS];
+        for (i, entry) in entries.iter().enumerate() {
+            user_data[0].push(big_uint_to_fp(entry.username_as_big_uint()));
+            for (j, balance) in entry.balances().iter().enumerate() {
+                user_data[j + 1].push(big_uint_to_fp(balance));
+            }
+        }
 
-        // After sending transaction of liability commitment, logs should be updated
-        liability_commitment_logs = summa_contract
-            .liabilities_commitment_submitted_filter()
-            .query()
-            .await?;
+        let mut round = Round::<N_CURRENCIES, N_POINTS, N_USERS>::new(
+            &signer,
+            advice_polys,
```

Comment:
advice_polys and user_data are representing exactly the same data in a slightly different way, so we're effectively duplicating the data. for a large user base this would cost a lot of memory. I think we need to have only one of the representations stored inside the round.